### PR TITLE
add block tests and fix minor bugs

### PR DESCRIPTION
**Pull request by yutongp**
basic block tests

### CODE REVIEW

File: disk/block_test.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+	"bytes"
+	"errors"
+	"io"
+	"os"
+	"strings"
+	"testing"
+)
+
+const (
+	tmpTestFile = "/tmp/cfs_disk_block_test"
```
Comment (xiang90) [inline disk/block_test.go]: can we use http://golang.org/pkg/io/ioutil/#TempFile?

File: disk/block_test.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+			_, err = f.Write(testFilePattern)
+		}
+		if err != nil {
+			t.Fatalf("can not write to test file %v", err)
+		}
+		length = length - int64(len(testFilePattern))
+	}
+
+	if err = f.Sync(); nil != err {
+		t.Fatal("can not sync test file: ", err)
```
Comment (xiang90) [inline disk/block_test.go]: inconsistent with line21

File: disk/block_test.go (hunk 1 above)
Comment (yutongp) [inline disk/block_test.go]: `TempFile` calls
`os.OpenFile(name, os.O_RDWR|os.O_CREATE|os.O_EXCL, 0600)`
But we do want to use `os.O_TRUNC` to reset the file every time we set up tests.

We can remove the file as well

Comment (xiang90) [inline disk/block_test.go]: well... at lease we can use the tmp dir. i do not want to use a fixed tmp dir.

File: disk/block_test.go [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+	"testing"
+)
+
+const (
+	tmpTestFile = "/tmp/cfs_disk_block_test"
+)
+
+var testFilePattern = []byte{'a', 'b', 'c', 'd', 'e', 'f', 'g'}
+
+func setUpTestFile(length int64, t *testing.T) *os.File {
```
Comment (xiang90) [inline disk/block_test.go:23]: can you add a few comments about why construct the test file this way?

File: disk/block_test.go [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+	n, err := readBlock(f, rp, 4, 20)
+	if err != io.EOF {
+		t.Errorf("expect error %v got %v", io.EOF, err)
+	}
+	if n != 6 {
+		t.Errorf("expect get data length = %v got %v", 6, n)
+	}
+}
+
+func TestBasicReadWriteBlock(t *testing.T) {
```
Comment (xiang90) [inline disk/block_test.go]: why this is basic? do we expect to have an advanced one? I hope we can have a more concrete name for the test too.

File: disk/block_test.go [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+		length = length - int64(len(testFilePattern))
+	}
+
+	if err = f.Sync(); nil != err {
+		t.Fatalf("can not sync test file: %v", err)
+	}
+	return f
+}
+
+func TestPartlyReadBlock(t *testing.T) {
```
Comment (xiang90) [inline disk/block_test.go:51]: i do not quite understand this test.

first we writes a block with index 8, block size 10.
then we tries to read a block with index 4, block 20.
1. the block size should be consistent between calls with same file.
2. we should expect a crc error in this case. not correct?

Comment (yutongp) [inline disk/block_test.go:51]: The first writeBlock is just to write in a correct crc(so I used a different blockSize and Index).
I want to test the last block partly read, so the file is 90 bytes, block size is 20 bytes, read index is 4. I expect read out 6 bytes and get a EOF

But currently we don't get a EOF.

Comment (xiang90) [inline disk/block_test.go:51]: well... it is a partial read, but the returned error should be a crc mismatch at least. io.eof is not an error, it just indicates the eof. so does this reveal a bug in our real code?

**Pull request merged**